src/Injector/ConditionalDiscoveryTrait.php

```diff
@@ -4,35 +4,26 @@
 
 namespace Laminas\ComponentInstaller\Injector;
 
+use Laminas\ComponentInstaller\ConfigDiscovery\DiscoveryInterface;
+
 use function str_replace;
 
+/**
+ * @psalm-require-extends AbstractInjector
+ */
 trait ConditionalDiscoveryTrait
 {
-    /**
-     * Prepends the package with a `\\` in order to ensure it is fully
-     * qualified, preventing issues in config files that are namespaced.
-     *
-     * @param string $package Package to inject into configuration.
-     * @param int $type One of the TYPE_* constants.
-     * @return bool
-     */
-    public function inject($package, $type)
+    public function inject(string $package, int $type): bool
     {
         if (! $this->validConfigAggregatorConfig()) {
             return false;
         }
 
+        /** @psalm-suppress ArgumentTypeCoercion Psalm has issues with trait inheritance and type parsing. */
```

Comment:
Uhh, this trait is problematic on multiple levels, if it uses `parent::` :|

Reply:
Anyway, I don't expect you to refactor it out. @weierophinney please less `trait`s overall (anywhere) :)

Reply:
It should already be quite fine due to the addition of `@psalm-require-extends`.

I guess that psalm simply does not fully parsed the types of the parent class and thus only sees the native type-hint but not the additional php docblock type-hint within the interface the parent class is implementing :-/